core/services/headtracker/head_saver.go

```diff
@@ -80,3 +66,13 @@ func (hs *headSaver) LatestChain() *eth.Head {
 func (hs *headSaver) Chain(hash common.Hash) *eth.Head {
 	return hs.heads.HeadByHash(hash)
 }
+
+var _ httypes.HeadSaver = &NullSaver{}
```

Comment:
Could consider exporting a singleton here and unexporting the implementation, e.g.:
```suggestion
var NullSaver httypes.HeadSaver = &nullSaver{}
```

Reply:
Done.